packages/next/build/swc/src/styled_jsx/mod.rs

```diff
@@ -0,0 +1,786 @@
+      }
+      _ => String::from("not_styled_jsx_tag"),
+    };
+    let style = if let JSXStyle::Local(style) = &styles[0] {
+      if tag != "resolve" {
+        self.external_hash = Some(hash_string(&style.hash.clone()));
+      }
+      style
+    } else {
+      bail!("This shouldn't happen, we already know that this is a template literal");
+    };
+    let css = transform_css(&style, tag == "global", &static_class_name)?;
+    if tag == "resolve" {
+      self.file_has_css_resolve = true;
+      return Ok(Expr::Object(ObjectLit {
+        props: vec![
+          PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+            key: PropName::Ident(Ident {
+              sym: "styles".into(),
+              span: DUMMY_SP,
+              optional: false,
+            }),
+            value: Box::new(Expr::JSXElement(Box::new(make_local_styled_jsx_el(
+              &style,
+              css,
+              &self.style_import_name.as_ref().unwrap(),
+            )))),
+          }))),
+          PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+            key: PropName::Ident(Ident {
+              sym: "className".into(),
+              span: DUMMY_SP,
+              optional: false,
+            }),
+            value: Box::new(class_name.unwrap()),
+          }))),
+        ],
+        span: DUMMY_SP,
+      }));
+    }
+    Ok(Expr::New(NewExpr {
+      callee: Box::new(Expr::Ident(Ident {
+        sym: "String".into(),
+        span: DUMMY_SP,
+        optional: false,
+      })),
+      args: Some(vec![ExprOrSpread {
+        expr: Box::new(css),
+        spread: None,
+      }]),
+      span: DUMMY_SP,
+      type_args: None,
+    }))
+  }
+
+  fn reset_styles_state(&mut self) {
+    self.has_styled_jsx = false;
+    self.static_class_name = None;
+    self.class_name = None;
+  }
+}
+
+fn is_styled_jsx(el: &JSXElement) -> bool {
+  if let JSXElementName::Ident(Ident { sym, .. }) = &el.opening.name {
+    if sym != "style" {
+      return false;
+    }
+  }
+
+  el.opening.attrs.iter().any(|attr| {
+    if let JSXAttrOrSpread::JSXAttr(JSXAttr {
+      name: JSXAttrName::Ident(Ident { sym, .. }),
+      ..
+    }) = &attr
+    {
+      if sym == "jsx" {
+        return true;
+      }
+    }
+    false
+  })
+}
+
+fn is_global(el: &JSXElement) -> bool {
+  if let JSXElementName::Ident(Ident { sym, .. }) = &el.opening.name {
+    if sym != "style" {
+      return false;
+    }
+  }
+
+  el.opening.attrs.iter().any(|attr| {
+    if let JSXAttrOrSpread::JSXAttr(JSXAttr {
+      name: JSXAttrName::Ident(Ident { sym, .. }),
+      ..
+    }) = &attr
+    {
+      if sym == "global" {
+        return true;
+      }
+    }
+    false
+  })
+}
+
+fn get_style_expr(el: &JSXElement) -> &Expr {
+  let non_whitespace_children: &Vec<&JSXElementChild> = &el
+    .children
+    .iter()
+    .filter(|child| {
+      if let JSXElementChild::JSXText(txt) = child {
+        if txt.value.to_string().chars().all(char::is_whitespace) {
```

Comment:
`.to_string()` is not required, as `JsWord` implements `Deref<Target = str>`.

(Note: String.chars() works because `String` implements Deref<Target = str>`.